healthcare/healthcare/doctype/patient_appointment/patient_appointment.py

```diff
@@ -418,7 +418,14 @@ def cancel_appointment(appointment_id):
 def cancel_sales_invoice(sales_invoice):
 	if frappe.db.get_single_value("Healthcare Settings", "automate_appointment_invoicing"):
 		if len(sales_invoice.items) == 1:
-			sales_invoice.cancel()
+			if sales_invoice.docstatus == 1:
+				frappe.throw(
+				_(
+					"The Sales Invoice {0} is Draft you cant cancel."
+				).format(
+					get_link_to_form("Sales Invoice", sales_invoice)
+				)
+			)
```

Comment:
What you're doing is throwing if document is submitted (docstatus == 1) right now. Do you mean to do something like the following?

```suggestion
			if sales_invoice.docstatus.is_submitted():
				sales_invoice.cancel()
```

Reply:
Exactly!

Reply:
I did the same in the first comment, but the response from akashkrishna619 shows a warning or notification to the user that there is a draft sales invoice

Reply:
Thanks, merging this.

Reply:
regards

Reply:
@akashkrishna619 
When will the merger take place? :)